L10n es aeat

agaldona opened the pull request:
Unificar campos y vistas.
Pasar código a la nueva API.
Nuevo atributo interno _aeat_number.

Commit:
[IMP] unificar campos en el módulo base + migrar a la nueva api.

pedrobaeza commented on l10n_es_aeat/README.rst:
```diff
@@ -0,0 +1,4 @@
+AEAT Base
+=========
+Módulo base para declaraciones de la AEAT.
+
```
Pon contributors, y formato de OCA según esta plantilla: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

pedrobaeza replied:
Indica un poco más la funcionalidad que incorpora (campos base, esqueleto para los modelos, secuencia automática, exportación básica...)

pedrobaeza commented on l10n_es_aeat/models/aeat_report.py:
```diff
@@ -0,0 +1,156 @@
+                                    self.company_id.partner_id.vat).groups()[1]
+        self.company_vat = comp_vat
+
+    company_id = fields.Many2one('res.company', string='Company',
```
Una forma de formatear esto que ocupa menos líneas es:

```
company_id = fields.Many2one(
    'res.company', string='Company', required=True, readonly=True,
    default=_default_company, states={'draft': [('readonly', False)]})
```

La clave está en no dejar ningún parámetro en la primera línea, y ya la segunda se indenta solo un nivel para adentro.

pedrobaeza replied:
Hazlo con todos los campos así de esta forma.

pedrobaeza commented on l10n_es_aeat/models/aeat_report.py:
```diff
@@ -0,0 +1,156 @@
+    def button_calculate(self):
+        res = self.calculate()
+        self.write({'state': 'calculated',
+                    'calculation_date': time.strftime(
```
Esto se puede poner ya `fields.Datetime.now()`

pedrobaeza commented on l10n_es_aeat/models/aeat_report.py:
```diff
@@ -0,0 +1,156 @@
+
+    @api.multi
+    def button_recalculate(self):
+        self.write({'calculation_date':
```
Ídem

pedrobaeza commented on l10n_es_aeat/models/aeat_report.py:
```diff
@@ -0,0 +1,156 @@
+
+    @api.multi
+    def unlink(self):
+        for item in self:
```
Puedes hacer un `if any(....): raise...`

pedrobaeza commented on l10n_es_aeat/models/aeat_report.py:
```diff
@@ -0,0 +1,156 @@
+            getattr(self, '_aeat_number')
+        except:
+            pass
+            # raise exceptions.Warning(
```
Esto lo descomentarás, no?

pedrobaeza commented on l10n_es_aeat/wizard/export_to_boe.py:
```diff
@@ -127,39 +119,81 @@ def _formatBoolean(self, value, yes='X', no=' '):
+        # Tipo de Registro
+        text += '1'
+        # Modelo Declaración
+        text += modelo_declaracion
```
Pon directamente `text += report._aeat_number`

pedrobaeza commented on l10n_es_aeat/wizard/export_to_boe.py:
```diff
@@ -167,37 +201,27 @@ def action_get_file(self, cr, uid, ids, context=None):
             time.strftime(_(DEFAULT_SERVER_DATE_FORMAT))
```
`fields.Date.today()`

pedrobaeza commented on l10n_es_aeat_mod340/models/mod340.py:
```diff
@@ -130,7 +126,6 @@ def _get_number_records(self, cr, uid, ids, field_name, args, context):
-        'calculation_date': fields.date('Calculation date', readonly=True),
     }
 
     _defaults = {
```
De aquí hay que eliminar el workflow que no sirve de nada.

Commit:
[IMP] 	README.rst mejorado
	mejora de código
	usabilidad del atributo _aeat_number
	secuencia automatica
	control al instanciar los modelos

Commit:
[IMP] quitar workflow al modelo 340 + correcciones necesarias para el correcto funcionamiento de los módulos

agaldona commented:
AEAT modelo 111.
- Ajustado a la nueva estructura del aeat
- Pasar a la nueva API
- Comprobar el cálculo de los datos + la exportación del fichero

pedrobaeza commented:
Ainara, ponlo por favor en PRs separados, aunque el del 111 incluya todos los commits de éste (es subirlo a otra rama y hacer el PR). Así, se podrán ver los cambios de uno y otro correctamente, o mergearlos por separado.

agaldona commented:
Ok. he pasado el modulo del modelo 111 a otra rama y la he quitado de aqui.

agaldona commented:
l10n_es_aeat
Motor de exportación genérica.
Motor calculo importes de códigos de impuesto.

pedrobaeza commented:
Ainara, por favor haz un rebase de todos los PRs, y así además se compilan en runbot.

agaldona commented:
Ya está!

pedrobaeza commented:
Errores flake8:

```
======== Testing test_flake8 ========
./l10n_es_aeat_mod340/__openerp__.py:82:8: E131 continuation line unaligned for hanging indent
./l10n_es_aeat_mod340/models/account.py:185:1: W391 blank line at end of file
======== Testing test_pylint ========
************* Module l10n-spain.l10n_es_aeat.wizard.export_to_boe
l10n_es_aeat/wizard/export_to_boe.py:32: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\$'. String constant might be missing an r prefix.
l10n_es_aeat/wizard/export_to_boe.py:32: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\{'. String constant might be missing an r prefix.
l10n_es_aeat/wizard/export_to_boe.py:32: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\}'. String constant might be missing an r prefix.
************* Module l10n-spain.l10n_es_aeat_mod130.models.mod130
l10n_es_aeat_mod130/models/mod130.py:140: [W0105(pointless-string-statement), L10nEsAeatMod130Report._get_periods] String statement has no effect
```

pedrobaeza commented:
Realizadas pruebas y pasado un tiempo prudencial, mergeo este cambio que es necesario para el resto de PR de modelos.
